### PR TITLE
Handled unsupported types in non master branch

### PR DESCRIPTION
**Pull request by PA-Rweins**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/36455

## Description
When running validate on non-master branch validate wont throw error for some files that are filtered.
Changed filtering so those types of files will raise an error while ignoring non-pack and test_data files

## Must have
- [x] Tests
- [x] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 1]
```diff
@@ -1103,6 +1103,45 @@ def test_is_there_spaces_in_the_end_of_name(self, pack: Pack, answer, dashboard_
...
+    def test_ignore_test_doc_non_pack_file_test_file(self):
+        """
+        Given
+            - File path
+        When
+            - File is part of the test_data directory
+        Then
+            - File is ignored and True is returned
+        """
+        file_path = 'Packs/SomeIntegration/IntegrationName/test_data/file.json'
```
Comment (yaakovi) [inline demisto_sdk/commands/validate/tests/validators_test.py:1128]: this is not a valid path for test data.
test data should appear under one of the following:
* `Packs/pack_id/test_data`
* `Packs/pack_id/Integrations/integration_id/test_data`
* `Packs/pack_id/Scripts/script_id/test_data`
* `Packs/pack_id/TestPlaybooks/test_data`

we can consider allowing freestyle location for the folder but it feels messy...

File: demisto_sdk/commands/validate/validate_manager.py [hunk 2]
```diff
@@ -1368,14 +1368,14 @@ def format_file_path(self, file_path, old_path, old_format_files):
 
         if file_path.split(os.path.sep)[0] in ('.gitlab', '.circleci', '.github'):
             return None
-
```
Comment (yaakovi) [inline demisto_sdk/commands/validate/validate_manager.py:1371]: revert deletion

File: demisto_sdk/commands/validate/validate_manager.py [hunk 3]
```diff
@@ -1402,6 +1402,15 @@ def format_file_path(self, file_path, old_path, old_format_files):
         else:
             return file_path
 
+    def ignore_test_doc_non_pack_file(self, file_path) -> bool:
```
Comment (yaakovi) [inline demisto_sdk/commands/validate/validate_manager.py:1405]: add type annotations

File: demisto_sdk/commands/validate/tests/validators_test.py [hunk 4]
```diff
@@ -1103,6 +1103,45 @@ def test_is_there_spaces_in_the_end_of_name(self, pack: Pack, answer, dashboard_
         res_validator = IntegrationValidator(structure)
         assert res_validator.is_there_spaces_in_the_end_of_name() is answer
 
+    def test_ignore_test_doc_non_pack_file_should_not_ignore(self):
```
Comment (yaakovi) [inline demisto_sdk/commands/validate/tests/validators_test.py:1106]: for all tests, add more testing cases (you can use `@parametrize`)

File: demisto_sdk/commands/validate/tests/validators_test.py (hunk 1 above)
Comment (PA-Rweins) [reply demisto_sdk/commands/validate/tests/validators_test.py:1128]: fixed

File: demisto_sdk/commands/validate/validate_manager.py (hunk 2 above)
Comment (PA-Rweins) [reply demisto_sdk/commands/validate/validate_manager.py:1371]: fixed

File: demisto_sdk/commands/validate/validate_manager.py (hunk 3 above)
Comment (PA-Rweins) [reply demisto_sdk/commands/validate/validate_manager.py:1405]: fixed

File: demisto_sdk/commands/validate/tests/validators_test.py (hunk 4 above)
Comment (PA-Rweins) [reply demisto_sdk/commands/validate/tests/validators_test.py]: Fixed

**Review by Bargenish — APPROVED** [review]

**Pull request merged**